region/maridia/outer/Crab Hole.json

```diff
@@ -1047,6 +1047,274 @@
+              "maxHeight": 1,
+              "minTiles": 5.4375,
+              "speedBooster": true,
+              "obstructions": [[1, 0]],
+              "requires": [
+                "canMomentumConservingMorph",
+                "canInsaneJump",
+                "canInsaneMidAirMorph"
+              ],
+              "note": [
+                "This applies to Warehouse Energy Tank Room.",
+                "It is double frame-perfect, requiring a last-frame jump in the other room just before Samus would get caught on the door ledge,",
+                "and a frame-perfect morph just before Samus would bonk the ceiling.",
+                "The aim-down must be delayed until at least 4 frames after entering the room."
+              ]
+            },
+            {
+              "minHeight": 1,
+              "maxHeight": 1,
+              "minTiles": 8.4375,
+              "speedBooster": "any",
+              "obstructions": [[1, 0]],
+              "requires": [
+                "canInsaneJump",
+                {"or": [
+                  "canMomentumConservingMorph",
+                  "canMomentumConservingTurnaround"
+                ]}
+              ],
+              "note": [
+                "This applies to Warehouse Entrance.",
+                "This requires a last-frame jump and specific horizontal alignment in order to avoid bonking the door frame.",
+                "A way to get a correct starting position is to jump, press against the ceiling overhang near the end of the runway,",
+                "land, turn around away from the door, then jump and mid-air turnaround toward the door."
+              ],
+              "devNote": [
+                "Speed Booster is not required for this."
+              ]
+            },
+            {
+              "minHeight": 1,
+              "maxHeight": 1,
+              "minTiles": 17,
+              "speedBooster": true,
+              "obstructions": [[1, 0]],
+              "requires": [
+                {"or": [
+                  "canMomentumConservingMorph",
+                  {"and": [
+                    "canMomentumConservingTurnaround",
+                    "canTrickyJump"
+                  ]}
+                ]}
+              ],
+              "note": [
+                "This applies to Warehouse Entrance.",
+                "This can be done in three ways:",
+                "1) Jump on the last possible frame in the previous room, turn around a little before hitting the ceiling, then morph and hold right to pass through the tunnel.",
+                "2) Perform a ceiling mockball through the transition,",
+                "unmorph a little before hitting the ceiling,",
+                "hold right to press against the wall (to avoid hitting the crab), then morph and hold right to pass through the tunnel.",
+                "3) Jump on the last possible frame in the previous room, then perform a ceiling mockball to go up through the hidden space."
```

Comment:
I hit the right crab doing this one.

Reply:
The trick is to release right while approaching the ceiling, like in this video: https://videos.maprando.com/video/5314. Added a note now to explain this.